src/pulumi-metadata.ts

```diff
@@ -72,17 +72,28 @@ export interface PulumiType {
 
 /**
  * The schema for the array items property
+ * A property will typically have either $ref, properties or additionalProperties
  */
 export interface PulumiPropertyItems {
     /**
      * A reference to another type
      */
     $ref?: string;
 
+    /**
+     * The properties of an object type
+     */
+    properties?: { [key: string]: PulumiPropertyItems };
+
     /**
      * The simple type (i.e. string, number, etc)
      */
     type?: string;
+
+    /**
+     * A type with additional properties
```

Comment:
This is called a Map type in Pulumi, I believe it would be helpful to model and call it as such. Maps are not objects. 

```
## Map Type

A reference to a map type. The "type" property must be set to "object" and the "additionalProperties" property may be present. No other properties may be present.

`object`

### Properties

---

#### `additionalProperties`

The element type of the map. Defaults to "string" when omitted.
```